File: ERPIN.Presentaion/src/app/Interfaces/Bases/InvoiceResponseBase.ts
import { InvoiceDetailsResponseBase } from './InvoiceDetailsResponseBase';

export interface InvoiceResponseBase {
  id: number;
  code: number;
  note: string | null;
  storeId: number | null;
  createdAt: string;
  paid: number;
  net: number;
  discount: number;
  discountRatio: number;
  tax: number;
  remain: number;
  total: number;
  invoiceDetails: InvoiceDetailsResponseBase[];
}

File: ERPIN.Services/DTOs/Bases/InvoiceResponseBase.cs
using ERPIN.Services.DTOs.Bases;

namespace ERPIN.Services.DTOs.Base;

public class InvoiceResponseBase
{
    public int Id { get; set; }
    public int Code { get; set; }
    public string? Note { get; set; }
    public DateTime CreatedAt { get; set; }
    public int? StoreId { get; set; }
    public decimal Paid { get; set; }
    public decimal Net { get; set; }
    public decimal Discount { get; set; }
    public decimal DiscountRatio { get; set; }
    public decimal Tax { get; set; }
    public decimal Remain { get; set; }
    public decimal Total { get; set; }

    public IList<InvoiceDetailResponseBase> InvoiceDetails { get; set; }
}
